contributors/design-proposals/multicluster/cluster-registry/api-design.md

```diff
@@ -0,0 +1,165 @@
+-   [Status](#status)
+-   [Auth](#auth)
+-   [Key differences vs existing Federation API `Cluster` object](#key-differences-vs-existing-federation-api-cluster-object)
+-   [Open questions](#open-questions)
+
+## Purpose
+
+The cluster registry API is intended to provide a common abstraction for other
+tools that will perform operations on multiple clusters. It provides an
+interface to a list of objects that will store metadata about clusters that can
+be used by other tools. The cluster registry implementation is meant to remain
+simple: we believe there is benefit in defining a common layer that can be used
+by many different tools to solve different kinds of multi-cluster problems.
+
+It may be helpful to consider this API as an extension of the `kubeconfig` file.
+The `kubeconfig` file contains a list of clusters with the auth data necessary
+for kubectl to access them; the cluster registry API intends to provide this
+data, plus some additional useful metadata, from a remote location instead of
+from the user's local machine.
+
+## Motivating use cases
+
+These were presented at the SIG-Federation F2F meeting on 8/4/17
+([Atlassian](https://docs.google.com/document/d/1PH859COCWSkRxILrQd6wDdYLGJaBtWQkSN3I-Lnam3g/edit#heading=h.suxgoa67n1aw),
+[CoreOS](https://docs.google.com/presentation/d/1InJagQNOxqA0ftK0peJLzyEFU2IZEXrJprDN6fcleMg/edit#slide=id.p),
+[Google](https://docs.google.com/presentation/d/1Php_HnHI-Sy20ieyd_jBgr7XTs0fKT0Cq9z6dC4zOMc/),
+[RedHat](https://docs.google.com/presentation/d/1dExjeSQTXI8_k00nqXRkSIFPTkzAzUTFtETU4Trg5yw/edit#slide=id.p)).
+Each of the use cases presented assumes the ability to access a registry of
+clusters, and so all are valid motivating use cases for the cluster registry
+API. Note that these use cases will generally require more tooling than the
+cluster registry itself. The cluster registry API will support what these other
+tools will need in order to operate, but will not intrinsically support these
+use cases.
+
+-   Consistent configuration across clusters/replication of resources
+-   Federated Ingress: load balancing across multiple clusters, potentially
+    geo-aware
+-   Multi-cluster application distribution, with policy/balancing
+-   Disaster recovery/failover
+-   Human- and tool- parseable interaction with a list of clusters
+-   Monitoring/health checking/status reporting for clusters, potentially with
+    dashboards
+-   Policy-based and jurisdictional placement of workloads
+
+## API
+
+This document defines the cluster registry API. It is an evolution of the
+[current Federation cluster API](https://github.com/kubernetes/kubernetes/blob/master/federation/apis/federation/types.go#L99),
+and is designed more specifically for the "cluster registry" use case in
+contrast to the Federation `Cluster` object, which was made for the
+active-control-plane Federation. All the names in the API are tentative and
+subject to change if we find better alternatives to describe those entities.
```

Comment:
I would remove this last sentence as part of this version of the doc. We already say below that the API itself is kept elsewhere.

Reply:
Good point, done.